core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -54,7 +54,12 @@ class TaskInfo(
    * accumulable to be updated multiple times in a single task or for two accumulables with the
    * same name but different IDs to exist in a task.
    */
-  val accumulables = ListBuffer[AccumulableInfo]()
+  lazy val accumulables: ListBuffer[AccumulableInfo] = ListBuffer(_accumulables: _*)
```

Comment:
Meh, kind of a toss-up. My original patch broke compatibility and I did this in 4c867f10ff5d8d80c7e8d44b4702785c0a017b64 to restore binary compatibility, but it risks changing semantics. I think I'll just revert 4c867f10ff5d8d80c7e8d44b4702785c0a017b64 and add the minor compatibility break to a lesser-used `@DeveloperAPI`.